subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedTaskExecutionIntegrationTest.groovy

```diff
@@ -485,4 +485,34 @@ class CachedTaskExecutionIntegrationTest extends AbstractIntegrationSpec impleme
         then:
         result.assertTaskSkipped(":cacheable")
     }
+
+    def "downstream task stays cached when upstream task is loaded from cache"() {
```

Comment:
So this is testing the interaction with the file system mirror? The output snapshots of the first task are not taken into account for the second task, only whatever is stored in the file system mirror, I guess, right?